### PR TITLE
boards: share the leftover mcuboot scratch flash between image 0 and image 1 for nRF devices

### PR DESCRIPTION
**Pull request by nvlsianpu**
Newest mcuboot uses move swap instead of scratch
for nRF devices

Share the leftover scratch flash between image 0 and image 1, or the storage partition.
For trustzone enabled devices, give the non secure
partitions the extra space.

Adjusted test overlays to above changes.

This PR supersedes #22228 as due the push to its branch during it was closed, GH GUI disallow to reopen it.

### CODE REVIEW

**Review by hakonfam — APPROVED** [review]
Should we add comments where not all of flash is used explaining why its like that? (primary and secondary have to be the same size)

File: boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts [hunk 1]
```diff
@@ -252,21 +252,12 @@ arduino_spi: &spi3 {
 		};
 		slot0_partition: partition@c000 {
 			label = "image-0";
-			reg = <0x0000C000 0x00067000>;
+			reg = <0x0000C000 0x000076000>;
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts:255]: @nvlsianpu please address the comments.

**Review by ioannisg — CHANGES_REQUESTED** [review]
This will not really work for TrustZone-capable nRF SoCs, where the flash partitions need to align with SPU boundaries.

**Comment by nvlsianpu** [comment]
>  This will not really work for TrustZone-capable nRF SoCs, where the flash partitions need to align with SPU boundaries.

Align as follow?
``` 
	default 0x8000 if SOC_NRF9160
	default 0x8000 if SOC_NRF5340_CPUAPP && NRF5340_CPUAPP_ERRATUM19
	default 0x4000 if SOC_NRF5340_CPUAPP
```

File: boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts (hunk 1 above)
[the remark this answers is not in the document]
Comment (nvlsianpu) [reply boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts]: @mniestroj fixed

File: boards/arm/nrf52840_blip/nrf52840_blip.dts [hunk 2]
```diff
@@ -143,21 +143,12 @@
 		};
 		slot0_partition: partition@c000 {
 			label = "image-0";
-			reg = <0x0000C000 0x00067000>;
+			reg = <0x0000C000 0x000076000>;
```
Comment (mniestroj) [inline boards/arm/nrf52840_blip/nrf52840_blip.dts:146]: Leading 0 to be removed.

File: boards/arm/nrf52840_mdk/nrf52840_mdk.dts [hunk 3]
```diff
@@ -169,21 +169,12 @@
 		};
 		slot0_partition: partition@c000 {
 			label = "image-0";
-			reg = <0x0000C000 0x00067000>;
+			reg = <0x0000C000 0x000076000>;
```
Comment (mniestroj) [inline boards/arm/nrf52840_mdk/nrf52840_mdk.dts:172]: Leading 0 to be removed.

File: boards/arm/nrf52840_papyr/nrf52840_papyr.dts [hunk 4]
```diff
@@ -141,21 +141,12 @@
 		};
 		slot0_partition: partition@c000 {
 			label = "image-0";
-			reg = <0x0000C000 0x00067000>;
+			reg = <0x0000C000 0x000076000>;
```
Comment (mniestroj) [inline boards/arm/nrf52840_papyr/nrf52840_papyr.dts:144]: Leading 0 to be removed.

**Comment by nvlsianpu** [comment]
@ioannisg I align changes to SPU boundaries. For nrf5340dk_nrf5340 applied the same alignment as for nrf5340pdk_nrf5340.

**Comment by nvlsianpu** [comment]
@mniestroj fixed

**Review by mniestroj — APPROVED** [review]

**Pull request closed without merging**